CMake/Findcereal.cmake

```
@@ -0,0 +1,53 @@
+#Findcereal.cmake
+find_path(CEREAL_INCLUDE_DIR
+  NAMES cereal
+  PATHS "$ENV{ProgramFiles}/cereal/include"
+  )
+
+if(CEREAL_INCLUDE_DIR)
+  # ------------------------------------------------------------------------
+  #  Extract version information from <CEREAL>
+  # ------------------------------------------------------------------------
+  set(CEREAL_FOUND YES)
+  set(CEREAL_VERSION_MAJOR 0)
+  set(CEREAL_VERSION_MINOR 0)
+  set(CEREAL_VERSION_PATCH 0)
+
+  if(EXISTS "${CEREAL_INCLUDE_DIR}/cereal/version.hpp")
+
+    # Read and parse cereal version header file for version number
+    file(READ "${CEREAL_INCLUDE_DIR}/cereal/version.hpp"
+        _CEREAL_HEADER_CONTENTS)
+    string(REGEX REPLACE ".*#define CEREAL_VERSION_MAJOR ([0-9]+).*" "\\1"
+        CEREAL_VERSION_MAJOR "${_CEREAL_HEADER_CONTENTS}")
+    string(REGEX REPLACE ".*#define CEREAL_VERSION_MINOR ([0-9]+).*" "\\1"
+        CEREAL_VERSION_MINOR "${_CEREAL_HEADER_CONTENTS}")
+    string(REGEX REPLACE ".*#define CEREAL_VERSION_PATCH ([0-9]+).*" "\\1"
+        CEREAL_VERSION_PATCH "${_CEREAL_HEADER_CONTENTS}")
+
+  elseif(EXISTS "${CEREAL_INCLUDE_DIR}/cereal/details/polymorphic_impl_fwd.hpp")
+
+    set(CEREAL_VERSION_MAJOR 1)
+    set(CEREAL_VERSION_MINOR 2)
+    set(CEREAL_VERSION_PATCH 0)
+  elseif(EXISTS "${CEREAL_INCLUDE_DIR}/cereal/types/valarray.hpp")
+
+    set(CEREAL_VERSION_MAJOR 1)
+    set(CEREAL_VERSION_MINOR 1)
+    set(CEREAL_VERSION_PATCH 2)
+  else()
+
+    set(CEREAL_VERSION_MAJOR 1)
+    set(CEREAL_VERSION_MINOR 1)
+    set(CEREAL_VERSION_PATCH 1) 
```

It seems the latest revision shows a bug with this script.  When an incorrect `CEREAL_INCLUDE_DIR` is given, here is the output:

```
CMake Error at /usr/local/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find cereal: Found unsuitable version "1.1.1", but required is at
  least "1.1.2" (found /home/vsts/work/1/s/build/cereal-1.3.0/include)
```

But in this case, no version is found at all.  So I think we should check that any cereal file at all is available.  So perhaps this:

```suggestion
elseif(EXISTS "${CEREAL_INCLUDE_DIR}/cereal/cereal.hpp")

  set(CEREAL_VERSION_MAJOR 1)
  set(CEREAL_VERSION_MINOR 1)
  set(CEREAL_VERSION_PATCH 1)
else()

  set(CEREAL_FOUND NO)
```